proto/iotics/api/search.proto

```diff
@@ -83,6 +92,111 @@ message SearchRequest {
   Range range = 20;
 }
 
+// AdvancedSearchRequest describes a search request with more filtering possibilities than SearchRequest. It returns
+// formatted details about the twins matched by the supplied filter.
+message AdvancedSearchRequest {
+  // Search request payload.
+  message Payload {
+
+    // Expected response type
+    ResponseType responseType = 1;
+
+    // The search filter, expressed as <LANG-TODO>
+    string filter = 2;
+  }
+
+  // Search request headers
+  Headers headers = 1;
+
+  // Search request scope
+  Scope scope = 2;
+
+  // Search request payload
+  Payload payload = 3;
+
+  // Search result limit (optional), affecting the maximum number of twins returned, per host. If not specified, no
+  // limit will be applied.
+  Limit limit = 4;
+}
+
+// AdvancedSearchRawRequest describes a search request with more filtering possibilities than SearchRequest. It returns
+// encoded RDF about the twins matched by the supplied filter.
+message AdvancedSearchRawRequest {
```

Comment:
This "duplicate" of `AdvancedSearchRequest` because of a single difference in specifying out format: `ResponseType` VS `RdfResultType`